.github/workflows/codeql-analysis.yml

```diff
@@ -14,11 +14,18 @@ on:
   schedule:
     - cron: '0 0 * * 3'
 
+permissions:
+  contents: read
+
 jobs:
   analyze:
     name: Analyze
     runs-on: ubuntu-latest
 
+    permissions:
+      # required for all workflow
```

Comment:
I don't quite understand this comment. Do you mean `all workflows`? Is there an "all" workflow?

Reply:
Sorry - this was lifted from their docs - "All" I think workflows and job might be conflated here. 
https://github.com/github/codeql-action/blob/main/.github/workflows/codeql.yml

You can see where they dogfood their own action how the security events `write` permission is set for build and version checks.

Reply:
I've removed the comment since you're right it is confusing using the "workflow" noun here